### PR TITLE
653 sidebar should highlight the selected layer

### PR DESCRIPTION
**Pull request by vanmeeuwen**

### CODE REVIEW

File: src/views/TopologyDisplayView.vue [hunk 1]
```diff
@@ -182,13 +182,15 @@ watch(
   () => props.nodeId,
   () => {
     if (props.nodeId) {
-      if (
-        typeof props.nodeId === 'string' &&
-        active.value[0] !== props.nodeId
+      if (typeof props.nodeId === 'string' && active.value !== props.nodeId) {
+        console.log('props.nodeid 0', props.nodeId)
```
Comment (hvangeffen) [inline src/views/TopologyDisplayView.vue]: Remove console logs

File: src/views/TopologyDisplayView.vue [hunk 2]
```diff
@@ -182,13 +182,15 @@ watch(
...
+        active.value = props.nodeId
+      } else if (
+        Array.isArray(props.nodeId) &&
+        active.value !== props.nodeId[0]
       ) {
-        active.value = [props.nodeId]
-      } else if (active.value[0] !== props.nodeId[0]) {
-        active.value = [props.nodeId[0]]
+        console.log('props.nodeid 1', props.nodeId[0])
```
Comment (hvangeffen) [inline src/views/TopologyDisplayView.vue]: Same here

File: src/services/useMenuItemsStack/index.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const _menuItems = toValue(menuItems)
+    const _activeItem = toValue(activeItem)
+
+    if (_activeItem !== '' && _menuItems.length > 0) {
+      const root: ColumnItem = {
+        id: 'rootNode',
+        name: '',
+        children: _menuItems,
+      }
+      const s = [root]
```
Comment (hvangeffen) [inline src/services/useMenuItemsStack/index.ts:22]: Could consider:
```suggestion
      stack.value = recursiveFind(root, _activeItem)
    }
  }
  return stack
}

function recursiveFind(item: ColumnItem, id: string): ColumnItem[] {
  if (item.id === id) return [item]

  for (const child of item.children ?? []) {
    const res = recursiveFind(child, id)
    if (res.length > 0) {
      return [item, ...res]
    }
  }

  return []
}

```

**Review by hvangeffen — APPROVED** [review]

File: src/services/useMenuItemsStack/index.ts (hunk 3 above)
Comment (vanmeeuwen) [reply src/services/useMenuItemsStack/index.ts:22]: Behavior in the column menu is slightly different, so I won't adopt this.

**Pull request merged**